src/main/java/io/knowledgeassets/myskills/server/download/FileDownloadController.java

```diff
@@ -0,0 +1,61 @@
+package io.knowledgeassets.myskills.server.download;
+
+import io.knowledgeassets.myskills.server.exception.UserProfileDocumentException;
+import io.knowledgeassets.myskills.server.user.profile.UserProfileService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.apache.poi.util.IOUtils;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+@Api(tags = "Download", description = "API to provide files for downloading.")
+@RestController
+@RequestMapping(path = "/download")
+public class FileDownloadController {
```

Comment:
Instead of this general class name I would recommend to use a more specific name for this business use case, e.g. `UserProfileDownloadController`. This better conveys the purpose of the class when looking at its name and promotes separation of concerns later on. If we really need some shared logic for providing file downloads we should inject a shared component into those separate controllers.

Reply:
I thought about it and the priorities of my personal feelings were 50/50. And I chose more general name. Thank you for the note. I changed the name.

Done.